lib/util/hostname.js

```diff
@@ -0,0 +1,26 @@
+import assert from 'assert';
+import statelessDNSServer from 'stateless-dns-server';
+
+export default function getHostname(config, expires) {
+  let statelessConfig = config.statelessHostname || {};
+  if (!statelessConfig || !statelessConfig.enabled) {
+    return config.host;
+  }
+
+  let secret = statelessConfig.secret || '1234';
+  let domain = statelessConfig.domain || '';
+  assert(secret, 'Must supply a secret for stateless dns server');
+  assert(domain, 'Must supply a domain name used for stateless hostname');
+
+  let hostname;
+  let ip = config.publicIp || '127.0.0.1';
```

Comment:
Shouldn't this `|| '127.0.0.1'` live in the place where publicIp is set... This seems a bit sketchy...